rats-rs/src/tee/sgx_dcap/attester.rs

```diff
@@ -34,13 +38,29 @@ impl GenericAttester for SgxDcapAttester {
             let mut sgx_report_data = sgx_report_data_t::default();
             sgx_report_data.d[..report_data.len()].clone_from_slice(report_data);
 
-            handler
-                .generate_quote(
-                    occlum_quote.as_mut_ptr(),
-                    &sgx_report_data as *const sgx_report_data_t,
-                )
-                .kind(ErrorKind::SgxDcapAttesterGenerateQuoteFailed)
-                .context("failed at generate_quote()")?;
+            let ptr = occlum_quote.as_mut_ptr() as usize;
+
+            #[cfg(feature = "async-tokio")]
+            {
+                task::spawn_blocking(move || {
+                    handler
+                        .generate_quote(
+                            ptr as *mut u8,
+                            &sgx_report_data as *const sgx_report_data_t,
+                        )
+                        .kind(ErrorKind::SgxDcapAttesterGenerateQuoteFailed)
+                        .context("failed at generate_quote()");
+                })
+                .await?;
+            }
+
+            #[cfg(not(feature = "async-tokio"))]
+            {
+                handler
+                    .generate_quote(ptr as *mut u8, &sgx_report_data as *const sgx_report_data_t)
+                    .kind(ErrorKind::SgxDcapAttesterGenerateQuoteFailed)
+                    .context("failed at generate_quote()");
+            }
```

Comment:
Some `Result`s should be handled here, don't forgot them.
```txt
warning: unused `std::result::Result` that must be used
  --> rats-rs/src/tee/sgx_dcap/attester.rs:59:17
   |
59 | /                 handler
60 | |                     .generate_quote(ptr as *mut u8, &sgx_report_data as *const sgx_report_data_t)
61 | |                     .kind(ErrorKind::SgxDcapAttesterGenerateQuoteFailed)
62 | |                     .context("failed at generate_quote()");
   | |__________________________________________________________^
   |
   = note: this `Result` may be an `Err` variant, which should be handled
   = note: `#[warn(unused_must_use)]` on by default
```

BTW, sharing raw pointer `ptr` between tasks is a bit dangerous. It's better to "move" the `Vec` between tasks.
```suggestion

            #[cfg(feature = "async-tokio")]
            {
                let handle = task::spawn_blocking(move || {
                    handler
                        .generate_quote(
                            occlum_quote.as_mut_ptr(),
                            &sgx_report_data as *const sgx_report_data_t,
                        )
                        .kind(ErrorKind::SgxDcapAttesterGenerateQuoteFailed)
                        .context("failed at generate_quote()")
                        .map(|_| occlum_quote)
                });
                occlum_quote = handle.await.context("the quote generation task panics")??;
            }

            #[cfg(not(feature = "async-tokio"))]
            {
                handler
                    .generate_quote(
                        occlum_quote.as_mut_ptr(),
                        &sgx_report_data as *const sgx_report_data_t,
                    )
                    .kind(ErrorKind::SgxDcapAttesterGenerateQuoteFailed)
                    .context("failed at generate_quote()")?;
            }
```